components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2734,7 +2734,29 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     public async getTeams(ctx: TraceContext): Promise<Team[]> {
         // Note: this operation is per-user only, hence needs no resource guard
         const user = await this.checkUser("getTeams");
-        return this.teamDB.findTeamsByUser(user.id);
+        const teams = await this.teamDB.findTeamsByUser(user.id);
+
+        // We need to check each team individually against our permission system.
+        // checks are promises, which resolve to { team, check } object
+        const checks = teams.map((team) =>
+            this.authorizer.check(ReadOrganizationInfo(user.id, team.id)).then((check) => ({ team, check })),
+        );
+        const checkResults = await Promise.allSettled(checks);
+
+        const accessibleTeams = [];
+        for (let result of checkResults) {
+            if (result.status !== "fulfilled") {
```

Comment:
we should log an error if the promise was rejected

Reply:
Sure, can do. What do we gain from this?

I'll first collect all rejected and log only once after the loop, if that's fine by you

Reply:
Would be good to log errors coming from `check` as we would optherwise silently swallow them and treat them as not authorized.
You can alternatively add a catch clause after the then in line 2742.

Reply:
> Would be good to log errors coming from check 
we already do log these on the `authorizer` level, but not necessarily in the context of the RPC (but the operation is included in the log)

Reply:
Ok, I couldn't tell looking at the code. In general, code that swallows possible exceptions is better to be avoided. We have linting for this but, of course, it is not intelligent enough to check all `allSettled` states.

Reply:
Added logging for the rejected case